pkg/generate/app/cmd/describe.go

```diff
@@ -171,11 +171,18 @@ func describeBuildPipelineWithImage(out io.Writer, ref app.ComponentReference, p
 				fmt.Fprintf(out, "      * The resulting image will be pushed to %s %q\n", to.Kind, to.Name)
 			}
 		}
-		if len(trackedImage) > 0 {
-			if noSource {
-				fmt.Fprintf(out, "      * Use 'start-build --from-dir=DIR' to trigger a new build\n")
-			} else {
+
+		if noSource {
+			// if we have no source, the user must always provide the source from the local dir(binary build)
+			fmt.Fprintf(out, "      * Use 'start-build --from-dir=DIR' to trigger a new build\n")
```

Comment:
yeah i didn't feel good about any choice i could make there.  honestly i wanted to just write "--from-XXXX" but that would be confusing too.

i don't really want to remove it from the start-build command because i think that's a helpful sample and frankly the most common way users will invoke it.

i might just change the warning to tell them to use --from-dir and trust that advanced users will realize there are other options.